plugins/playstation/src/playstation-header.vala

```diff
@@ -0,0 +1,87 @@
+// This file is part of GNOME Games. License: GPLv3
+
+private class Games.PlayStationHeader : Object {
+  private File cue_file;
+  private File bin_file;
+	private string parent_directory;
+
+	private const size_t[] OFFSETS = { 0xD368, 0xD3A8 };
+	private const string[] IDS = { "SLUS", "SCUS", "SLES", "SCES", "SLPS", "SLPM", "SCPS" };
+
+	private size_t? header_offset;
+
+  public PlayStationHeader (File cue_file) {
+    this.cue_file = cue_file;
+  }
+
+  public void check_validity () throws PlayStationError {
+    check_cue_info ();
+
+		foreach (var offset in OFFSETS) {
+			var data = get_string_at_offset (offset, 50);
+			if (data.contains ("BOOT")) {
+				foreach (var id in IDS) {
```

Comment:
Rather than relying on a list of possible values, what about using a regex, like this:
```
var id_regex = /cdrom:\\?(.*?);/;

MatchInfo match_info;
if (!regex.match (data, RegexMatchFlags.ANCHORED, out match_info))
    continue;

var raw_id = match_info.fetch (1);
raw_id = raw_id.replace (".", "");
raw_id = raw_id.replace ("_", "-");
```